### PR TITLE
Implemented basic group syncing from azure to orchard on login

### PR DESCRIPTION
**Pull request by wharri220**
Cleaned up Orchard.Azure.Authentication module and implemented group syncing for any azure groups that match orchard roles for the signed in user.

### CODE REVIEW

**Comment by sebastienros** [comment]
I have forced push a squash commit on top of dev and fixed the merge issues. Your PR was in a bad state (you merged dev into it and it was in a bad state after).

Can you ensure that what is in the azureauth branch is correct?

**Comment by wharri220** [comment]
Yea, I'll check it out. I attempted to rebase from dev, and got several conflicts. Wasn't sure what was needed and what wasn't, so I merged. Thanks for fixing. I'll verify the azureauth branch.

**Comment by wharri220** [comment]
It looks good except for one problem in the web.config for Orchard.Web. It's referencing assembly Microsoft.Owin.Host.SystemWeb version="3.0.0.0". This needs to be version="3.0.1.0". I was getting an assembly reference error. I fixed it locally and everything runs as expected. Should I submit another PR for that fix?

**Comment by sebastienros** [comment]
PR please

File: src/Orchard.Web/Modules/Orchard.Azure.Authentication/Migrations/AzureActiveDirectorySyncPartMigrations.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+using System;
+
+namespace Orchard.Azure.Authentication.Migrations
+{
+	public class AzureActiveDirectorySyncPartMigrations : DataMigrationImpl
+	{
+
+		public int Create()
+		{
+			SchemaBuilder.CreateTable(typeof(AzureActiveDirectorySyncPartRecord).Name,
```
Comment (Ermesx) [inline src/Orchard.Web/Modules/Orchard.Azure.Authentication/Migrations/AzureActiveDirectorySyncPartMigrations.cs]: Here should be string. What if someone change that class name ?

Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Azure.Authentication/Migrations/AzureActiveDirectorySyncPartMigrations.cs]: Correct, this is what we do in migrations, even with C# 6 we don't use `nameof`.

File: src/Orchard.Web/Modules/Orchard.Azure.Authentication/OwinMiddlewares.cs [hunk 2]
```diff
@@ -13,71 +16,110 @@
...
+            }
+            catch (Exception ex)
+            {
+                Logger.Log(LogLevel.Debug, ex, "An error occured while accessing azure settings: {0}");
+            }
         }
 
-        public IEnumerable<OwinMiddlewareRegistration> GetOwinMiddlewares() {
+        public IEnumerable<OwinMiddlewareRegistration> GetOwinMiddlewares()
+        {
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Azure.Authentication/OwinMiddlewares.cs]: Be careful with the formatting

File: src/Orchard.Web/Modules/Orchard.Azure.Authentication/Web.config [hunk 3]
```diff
@@ -34,59 +34,59 @@
...
-        <assemblyIdentity name="Microsoft.Owin" publicKeyToken="31bf3856ad364e35" culture="neutral"/>
-        <bindingRedirect oldVersion="0.0.0.0-3.0.1.0" newVersion="3.0.1.0"/>
+        <assemblyIdentity name="Microsoft.Owin" publicKeyToken="31bf3856ad364e35" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-3.0.1.0" newVersion="3.0.1.0" />
       </dependentAssembly>
       <dependentAssembly>
-        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral"/>
-        <bindingRedirect oldVersion="0.0.0.0-9.0.0.0" newVersion="9.0.0.0"/>
+        <assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30ad4fe6b2a6aeed" culture="neutral" />
+        <bindingRedirect oldVersion="0.0.0.0-7.0.0.0" newVersion="7.0.0.0" />
```
Comment (sebastienros) [inline src/Orchard.Web/Modules/Orchard.Azure.Authentication/Web.config]: Any reason

File: src/Orchard.Web/Modules/Orchard.Azure.Authentication/OwinMiddlewares.cs (hunk 2 above)
Comment (wharri220) [inline src/Orchard.Web/Modules/Orchard.Azure.Authentication/OwinMiddlewares.cs]: I thought I had taken care of reformatting everything. But apparently not, will get that fixed.

File: src/Orchard.Web/Modules/Orchard.Azure.Authentication/Migrations/AzureActiveDirectorySyncPartMigrations.cs (hunk 1 above)
Comment (wharri220) [inline src/Orchard.Web/Modules/Orchard.Azure.Authentication/Migrations/AzureActiveDirectorySyncPartMigrations.cs]: Hmm, this migration class is no longer used. It was supposed to have been removed from module anyway. So I will remove it if it still exists in the code.

**Pull request merged**